### PR TITLE
Adds awesome azd as default template source when not defined

### PR DESCRIPTION
**Pull request by wbreza**
Adds `awesome-azd` as the default template source when template sources has not been defined.

- When `template.sources` is not defined in `config.json` => Add `awesome-azd`
- When `template.sources` is empty => Empty template sources

## Use case: 1st run experience after release

`template.sources` will not be defined, `awesome-azd` will automatically get added and saved

## Use case: User manually removes all templates sources

User has used `azd template source remove <key>` for all registered template sources.  New sources will need to be manually registered for anything to show up in `azd template list` or `azd init` experience.

> [!NOTE] 
> `Minimal` will still show in `azd init` experience with empty template sources.

## Use case: User manually clears `config.json` or `azd config reset`

Fallback to 1st run experience above where template sources have not been defined.

### CODE REVIEW

**Comment by weikanglim** [comment]
To summarize my understanding of the experience:

1. `awesome-azd` is a configurable option. Users can opt-out by `azd template source remove awesome-azd`.
2. Otherwise, it's expected that `awesome-azd` will show up in `config.json`, which includes after `azd config reset`. The way it shows up is technically when you run anything that "lists templates", it gets lazily added. This is an internal implementation detail that doesn't matter unless the user cares about having a "stable" `config.json`, or that user touches `config.json`, in which case the user needs understanding that `template.sources == nil` is not the same as `template.sources == []`.

I think the design has definite merits in it uses the extensible feature it was built upon. I could also see a more streamlined (but perhaps less educating) experience where:

1. `awesome-azd` is a feature that is always enabled. And there is a config key `azd config set template.show.awesome-azd false` that changes the behavior. It's not treated as an extensible "template source" feature.

The good part of the design is that it avoids the config complication and defines the concrete default behavior in a straightforward way for product maintainers and also for users. But perhaps this is a minutia and the exposure of `template source` itself is the bigger ticket item.

File: cli/azd/pkg/templates/source_manager.go
Comment (weikanglim) [inline cli/azd/pkg/templates/source_manager.go:99]: Does `default` now not mean the "default source'?

**Comment by jongio** [comment]
@savannahostrowski - Loading community templates by default might lead to confusion by end user on which templates are MS vs non-MS.  Also, there might be security risks involved, etc.  Time to intro a disclaimer to this command?

**Comment by savannahostrowski** [comment]
Yeah - we can definitely add a notice to the top of the table output here to indicate that there are risks running 3rd party code (we already include this notice on init too).

File: cli/azd/pkg/templates/source_manager.go
Comment (wbreza) [reply cli/azd/pkg/templates/source_manager.go:99]: Correct - default is more like a **legacy** template source now.  @savannahostrowski any thoughts on how we would want to reference this if users wanted to go back to our manually curated list?

Comment (savannahostrowski) [reply cli/azd/pkg/templates/source_manager.go:99]: Hmmm, that's an interesting question. I could see us still having them labelled as default. I don't see a name that's better suited off the top of my head.

**Review by ellismg — APPROVED** [review]
Changes LGTM if the others are happy with where the default conversation landed.  I also don't have a great alternative to propose.

**Review by weikanglim — APPROVED** [review]
Sorry, didn't get around to click the approve button. Didn't have any blocking comments,

**Pull request merged**
